lib/vssh/libssh2.c

```diff
@@ -3173,6 +3179,11 @@ static CURLcode ssh_connect(struct connectdata *conn, bool *done)
 #ifdef HAVE_LIBSSH2_KNOWNHOST_API
   if(data->set.str[STRING_SSH_KNOWNHOSTS]) {
     int rc;
+
+    if(data->share && (data->share->specifier & (1<< CURL_LOCK_DATA_KNOWNHOSTS)) ) {
+      Curl_share_lock(data, CURL_LOCK_DATA_KNOWNHOSTS, CURL_LOCK_ACCESS_SINGLE);
+      ssh->kh = data->share->known_hosts_shared;
+    }
     ssh->kh = libssh2_knownhost_init(ssh->ssh_session);
```

Comment:
`ssh->kh` gets overwritten here even if assigned in the block just above. Is that really intended?